test/blackbox-tests/test-cases/lib/run.t

```diff
@@ -0,0 +1,129 @@
+----------------------------------------------------------------------------------
+Testsuite for the %{lib...} and %{lib-private...} variable.
+TODO: Fix %{libexec} and %{libexec-private} variables and test them.
+
+  $ cat >sdune <<'EOF'
+  > #!/usr/bin/env bash
+  > DUNE_SANDBOX=symlink dune "$@"
+  > EOF
+  $ chmod +x sdune
+
+----------------------------------------------------------------------------------
+* Find a public library using the %{lib:...} variable
+
+  $ echo "(lang dune 2.0)" > dune-project
+  $ mkdir -p src
+  $ cat >src/dune <<EOF
+  > (library
+  >  (name private_lib)
+  >  (public_name public_lib)
+  >  (modules a))
+  > EOF
+
+  $ cat >dune <<EOF
+  > (rule
+  >  (alias find-a)
+  >  (action (echo "%{lib:public_lib:a.ml}")))
+  > EOF
+
+  $ cat >src/a.ml <<EOF
+  > let a = "A"
+  > EOF
+  $ cat >src/a.mli <<EOF
+  > val a : string
+  > EOF
+
+  $ touch public_lib.opam
+  $ ./sdune build @find-a
+  ../install/default/lib/public_lib/a.ml
+
+----------------------------------------------------------------------------------
+* Error when finding a public library by its private name using the %{lib:...} variable
+
+  $ cat >src/dune <<EOF
+  > (library
+  >  (name private_lib)
+  >  (public_name public_lib)
+  >  (modules a))
+  > EOF
+
+  $ cat >dune <<EOF
+  > (rule
+  >  (alias find-a)
+  >  (action (echo "%{lib:private_lib:a.ml}")))
+  > EOF
+
+  $ ./sdune build @find-a
+  File "dune", line 3, characters 18-39:
+  3 |  (action (echo "%{lib:private_lib:a.ml}")))
+                        ^^^^^^^^^^^^^^^^^^^^^
+  Error: Library "private_lib" is not public. You can use the "lib" variable
+  only with public libraries that are installed.
```

Comment:
This makes me think that it is possible to have public libraries that are not installed.

Reply:
Changed to: 

```
Error: The library "private_lib" is not public. The variable "lib" expands to
the file's installation path which is not defined for private libraries.
```

Hopefully, it is clearer.

Reply:
I'm hitting this error 1.5 years later and I'm confused: why do we say the library is private even though it's actually public? At the very least the error message is confusing, but I can't even tell why it's an error in the first place.